File: Data Structures & Algorithms/house-robber/submission-0.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        memo=[]
        def robh(ints):
            ans=0
            if len(ints) <= len(memo):
                return memo[len(ints)-1]
            if len(ints)==1:
                ans=ints[0]
            if len(ints)==2:
                ans=max(ints[0],ints[1])
            if len(ints)>2:
                ans=max(robh(ints[:len(ints)-2])+ints[-1],robh(ints[:len(ints)-1]))
            if len(ints)==len(memo)+1:
                memo.append(ans)
            return ans
        return robh(nums)

File: Data Structures & Algorithms/longest-increasing-subsequence/submission-0.cpp
class Solution {
public:
    int lengthOfLIS(vector<int>& nums) {
        int ans=1;
        vector<int> lengths;
        lengths.resize(nums.size(), 1);
        for (int i=1;i<nums.size();++i) {
            for (int j=0; j<i; ++j) {
                if (nums[i]>nums[j]) {
                    lengths[i]=max(lengths[i], lengths[j]+1);
                    ans=max(ans,lengths[i]);
                }
            }
        }
        return ans;
    }
};
